sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -572,6 +572,14 @@ object SQLConf {
       "disable logging or -1 to apply no limit.")
     .createWithDefault(1000)
 
+  val WHOLESTAGE_MAX_LINES_PER_FUNCTION = buildConf("spark.sql.codegen.maxLinesPerFunction")
+    .internal()
+    .doc("The maximum lines of a single Java function generated by whole-stage codegen. " +
+      "When the generated function exceeds this threshold, " +
+      "the whole-stage codegen is deactivated for this subtree of the current query plan.")
+    .intConf
+    .createWithDefault(1500)
```

Comment:
I think that this value depends on what code is generated by the whole-stage codegen for each query. In other words, when Java byte code per line is larger than 6 (= `8192 / 1500`) in a generated Java code, `1500` would not work.
@eatoncys why are you sure that this can apply to other Java programs?